c++/nda/algorithms.hpp

```diff
@@ -105,16 +105,13 @@ namespace nda {
   // --------------- Computation of the matrix norm ------------------------
 
   /**
-   * @tparam A Anything modeling NdArray
-   * @param a The object of type A
-   * @return The minimum element of A
+   * @tparam A Anything modeling the ArrayOfRank<2> concept
+   * @param m The object of type A
    */
-  template <typename T>
-  // FIxme IsDoubleOrComplex
-  //require( is_real_or_complex<T>)
-  double frobenius_norm(matrix<T> const &a) {
+  template <CONCEPT(ArrayOfRank<2>) A>
+  double frobenius_norm(A const &a) REQUIRES17(is_ndarray_v<A> &&has_rank<A, 2>) {
     return std::sqrt(fold(
-       [](double r, T const &x) -> double {
+       [](double r, get_value_t<A> const &x) -> double {
```

Comment:
Use
auto const &
it is much more readable, one does not need to check that the type is correct, 
i.e. that **no silent conversion will ever occur...**

Reply:
Yes I agree here

Reply:
why is it different from the previous case ? ok the trait is more complex to read though...
It is a matter of style anyway